src/LetsEncrypt/Internal/CertificateFactory.cs

```diff
@@ -115,66 +123,61 @@ private void EnsureAgreementToTermsOfServices(Uri tosUri)
             throw new InvalidOperationException("Could not automatically accept the terms of service");
         }
 
-        private async Task ValidateDomainOwnershipAsync(string domainName, CancellationToken cancellationToken)
+        private async Task ValidateDomainOwnershipAsync(IAuthorizationContext authorizationContext, CancellationToken cancellationToken)
         {
             cancellationToken.ThrowIfCancellationRequested();
 
+            var authorization = await authorizationContext.Resource();
+            var domainName = authorization.Identifier.Value;
+
             _logger.LogDebug("Requesting authorization to create certificate for {domainName}", domainName);
-            var auth = await _client.NewAuthorization(new AuthorizationIdentifier
-            {
-                Type = AuthorizationIdentifierTypes.Dns,
-                Value = domainName,
-            });
-            _logger.LogResponse("NewAuthorization", auth);
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var httpChallenge = auth.Data.Challenges.FirstOrDefault(c => c.Type == ChallengeTypes.Http01);
+            var httpChallenge = await authorizationContext.Http();
+
+            cancellationToken.ThrowIfCancellationRequested();
 
             if (httpChallenge == null)
             {
                 throw new InvalidOperationException($"Did not receive challenge information for challenge type {ChallengeTypes.Http01}");
             }
 
-            var keyAuth = _client.ComputeKeyAuthorization(httpChallenge);
+            var keyAuth = httpChallenge.KeyAuthz;
             _challengeStore.AddChallengeResponse(httpChallenge.Token, keyAuth);
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            _logger.LogDebug("Requesting completion of challenge to prove ownership of {domainName}", domainName);
-
-            var challengeCompletion = await _client.CompleteChallenge(httpChallenge);
+            _logger.LogDebug("Requesting completion of challenge to prove ownership of domain {domainName}", domainName);
 
-            _logger.LogResponse("CompleteChallenge", challengeCompletion);
+            var challange = await httpChallenge.Validate();
```

Comment:
```suggestion
            var challenge = await httpChallenge.Validate();
```
Nit: typo correction.